torch/csrc/jit/passes/op_replacement.cpp

```diff
@@ -0,0 +1,69 @@
+#include <c10/util/Exception.h>
+#include <torch/csrc/jit/ir/irparser.h>
+#include <torch/csrc/jit/operator_upgraders/upgraders.h>
+#include <torch/csrc/jit/operator_upgraders/utils.h>
+#include <torch/csrc/jit/operator_upgraders/version_map.h>
+#include <torch/csrc/jit/passes/op_replacement.h>
+#include <torch/csrc/jit/runtime/graph_iterator.h>
+#include <limits>
+#include <regex>
+#include <string>
+
+namespace torch {
+namespace jit {
+
+struct OpsReplacer {
+  OpsReplacer(std::shared_ptr<Graph> graph) : graph_(std::move(graph)) {}
+
+  void run() {
+    if (!graph_->get_op_version().has_value()) {
+      return;
+    }
+    auto current_version = graph_->get_op_version().value();
+    DepthFirstGraphNodeIterator graph_it(graph_);
+    Node* node = graph_it.next();
+    while (node) {
```

Comment:
Nit:
It is actually safer to use a for loop instead of `while` here to avoid accidental early-exit-forgot-to-go-next issue.